File: ListViewDemo/app/src/main/java/com/gbreed/listviewdemo/MainActivity.java
package com.gbreed.listviewdemo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;

import static java.util.Arrays.asList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ListView listFriends = findViewById(R.id.listView);

        final ArrayList<String> friends = new ArrayList<>(asList("Joe", "John", "Joey", "Ross", "Marcus"));

        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, friends);

        listFriends.setAdapter(adapter);

        listFriends.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Log.i("Name: ", friends.get(position));
                Toast.makeText(MainActivity.this, "" + friends.get(position), Toast.LENGTH_SHORT).show();
            }
        });
    }

}

File: KotlinFun/app/src/main/java/com/gbreed/kotlinfun/MainActivity.kt
package com.gbreed.kotlinfun

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity()
{
    var count = 0

    override fun onCreate(savedInstanceState: Bundle?)
    {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        count = 0
        var textView = findViewById<TextView>(R.id.textCount)
        textView.setText(count.toString())
    }

    fun resetNum(view : View)
    {
        count = 0
        textCount.setText(count.toString())
    }

    fun addNum(view : View)
    {
        count += 1
        textCount.setText(count)
    }
}
